test/production/eslint/test/next-build-and-lint.test.ts

```diff
@@ -58,8 +58,8 @@ describe('Next Build', () => {
             parser,
             settings,
           }).toEqual({
-            // parser: require.resolve('eslint-config-next')
-            parser: expect.stringContaining('eslint-config-next'),
+            // parser: require.resolve('@typescript-eslint/parser')
+            parser: expect.stringContaining('@typescript-eslint/parser'),
```

Comment:
Is this intended @JoshuaKGoldberg ? Seems like `eslint-config-next/parser` is now unused?

Reply:
I think so, but I don't follow exactly why to use it in the first place. My understanding is that it's made so that `next`'s custom parser is used for JS files (`next/dist/compiled/babel/eslint-parser`) but I don't know what that gives?

For users who aren't using TypeScript, I don't believe this PR changes anything?

Reply:
> For users who aren't using TypeScript, I don't believe this PR changes anything?

It does since it previously uses `eslint-config-next` as the parser previously and now uses `@typescript-eslint/parser` for the JS files. 

I'll try to find why we needed the extra parser and will schedule it for removal after 15.x. In the meantime, can you double check how this behavior changed? I'd assume that this PR should not affect .js files in any way.

Reply:
It was a misplaced `)` on my end. https://github.com/vercel/next.js/pull/52845/files#r1653012693 -> https://github.com/vercel/next.js/pull/52845/commits/40ba2cea309fcf17063b9d23ed4e020bd4494c49.

Reply:
@eps1lon I don't know how to investigate the new failures, but I think these unit tests & their tested behavior are fixed up.

Reply:
Looks good now 👍🏻